docs/DeveloperGuide.adoc

```diff
@@ -837,14 +837,311 @@ Use case ends.
+
+*MSS*
+
+1. User requests to list persons
+2. AddressBook shows a list of persons
+3. User requests to add notes to specific person in the list
+4. AddressBook adds notes to the specified person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+[none]
+* 3a. The given index is invalid
++
+[none]
+** 3a1. AddressBook shows an error message
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Add contact specific tasks
+
+*MSS*
+
+1.  User selects contact to add a task for
+2.  AddressBook prompts user to enter the task details
+3.  User enters task description and task deadline
+4.  AddressBook adds task to specified contact and shows a success message.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. User selects invalid contact
++
+[none]
+** 1a1. AddressBook shows an error message
++
+Use case ends.
+
+[none]
+* 3a. User enters invalid task details
++
+[none]
+** 3a1. AddressBook shows en error message
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Find persons by name
+
+*MSS*
+
+1.	User requests to find persons with specific names
+2.	AddressBook shows list of persons with specified names
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. User does not specify any name
++
+[none]
+** 1a1. AddressBook shows an error message
++
+Use case ends.
+
+[none]
+* 1b. AddressBook does not have any names specified by user
++
+** 1b1. AddressBook shows empty list
++
+Use case ends.
+
+[discrete]
+=== Use case: Find persons by tags
+
+*MSS*
+
+1.	User requests to find persons with specific tags
+2.	AddressBook shows list of persons with specified tags
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. User does not specify any tag
++
+[none]
+** 1a1. AddressBook shows an error message
++
+Use case ends.
+
+[none]
+* 1b. AddressBook does not have any tags specified by user
++
+[none]
+** 1b1. AddressBook shows empty list
++
+Use case ends.
+
+[discrete]
+=== Use case: Remove specific tag
+
+*MSS*
+
+1.  User requests to remove a tag
+2.  AddressBook shows a list of all the tags used in the application
+3.  User enters the tag they want to remove
+4.  AddressBook removes the specific tag from all contacts who have this tag assigned to them
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There are no tags
++
+[none]
+** 2a1. AddressBook shows an error message
++
+Use case ends.
+
+[none]
+* 3a. User enters invalid tag
++
+[none]
+** 3a1. AddressBook shows an error message
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Export contacts
+
+*MSS*
+
+1.  User selects contacts to export from the current list of contacts
+2.  AddressBook stores the selection and processes the contacts to export
+3.  User requests to create a file containing the selected contacts
+4.  AddressBook creates the file and adds in contacts
+5.  User specifies location of the file to be saved
+6.  AddressBook saves the file in the location specified and shows a success message
++
```

Comment:
Perhaps for steps 2 and 4, instead of stating what AddressBook has done (which cannot be seen by user), you can say that AddressBook shows a success message instead?

Reply:
Alright, will update